resallocserver/manager.py

```diff
@@ -298,6 +304,55 @@ def job(self):
         self.event.set()
 
 
+class CleanUnknownWorker(Worker):
+    """
+    Delete all resources that are not recognized by resalloc
+    https://github.com/praiskup/resalloc/issues/88
+    """
+    def job(self):
+        if not self.pool.cmd_list:
+            return
+
+        all_resources = self._list_all_resources()
+        known_resources = self._list_known_resources()
+        unknown_resources = set(all_resources) - set(known_resources)
+
+        # There is no assigned resource ID for the given resource, that's
+        # the whole point. Let's simply use 0, which means the termination
+        # output for all unknown resources will be logged into
+        # hooks/000000_terminate
+        res_id=0
+
+        for resource in unknown_resources:
+            run_command(
+                self.pool.id,
+                res_id,
+                resource,
+                None,
+                self.pool.cmd_delete,
+                'terminate',
+                data=None,
+            )
+
+    def _list_all_resources(self):
+        """
+        List all resources using `cmd_list` from `pools.yaml`. This may include
+        resources that are not known to resalloc
+        """
+        env = command_env(pool_id=self.pool.id)
+        cmd = [self.pool.cmd_list]
+
+        proc = subprocess.Popen(cmd, env=env, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
```

Comment:
Hm, I missed this before -- btw. why not to use `run_command()` directly?

Reply:
I suppose there is nothing that would block us from using `run_command`. We would call it with `res_id=0` and `ltype='list'`?

Reply:
Perhaps ltype=pool.name+"_list"?

Reply:
That will create a log file called `000000__list` which is a bit weird.
What if we did `ltype=None` and simply didn't log the output?

Reply:
IMO it is still better to have there something, even `ltype='list'` as you originally proposed.  This logging might make debugging **significantly** easier.  Pity we can't use the pool name in the filename to avoid the interleaved outputs. But still good.

Reply:
Updated